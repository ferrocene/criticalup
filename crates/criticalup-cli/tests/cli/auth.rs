// SPDX-FileCopyrightText: The Ferrocene Developers
// SPDX-License-Identifier: MIT OR Apache-2.0

use crate::assert_output;
use crate::utils::{TestEnvironment, MOCK_AUTH_TOKENS};

#[tokio::test]
async fn help_message() {
    let test_env = TestEnvironment::prepare().await;
    assert_output!(test_env.cmd().args(["auth", "--help"]));
}

#[tokio::test]
async fn no_token() {
    let test_env = TestEnvironment::prepare().await;

    assert_output!(test_env.cmd().arg("auth"));
    assert_eq!(0, test_env.requests_served_by_mock_download_server().await);
}

#[tokio::test]
async fn invalid_token() {
    let test_env = TestEnvironment::prepare().await;
    set_token(&test_env, MOCK_AUTH_TOKENS[2].0).await;
    test_env.revoke_token(MOCK_AUTH_TOKENS[2].0).await;

    assert_output!(test_env.cmd().arg("auth"));
    assert_eq!(2, test_env.requests_served_by_mock_download_server().await);
}

#[tokio::test]
async fn token_without_expiry() {
    let test_env = TestEnvironment::prepare().await;
    set_token(&test_env, MOCK_AUTH_TOKENS[0].0).await;

    assert_output!(test_env.cmd().arg("auth"));
    assert_eq!(2, test_env.requests_served_by_mock_download_server().await);
}

#[tokio::test]
async fn token_with_expiry() {
    let test_env = TestEnvironment::prepare().await;
    set_token(&test_env, MOCK_AUTH_TOKENS[1].0).await;

    assert_output!(test_env.cmd().arg("auth"));
    assert_eq!(2, test_env.requests_served_by_mock_download_server().await);
}

async fn set_token(test_env: &TestEnvironment, token: &str) {
    let output = test_env
        .cmd()
        .args(["auth", "set", token])
        .output()
        .await
        .expect("failed to set the token");
    assert!(output.status.success());
}
