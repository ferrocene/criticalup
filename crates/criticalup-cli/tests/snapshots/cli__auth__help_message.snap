---
source: crates/criticalup-cli/tests/cli/auth.rs
expression: repr
---
exit: exit status: 0

empty stdout

stderr
------
Show and change authentication with the download server

Usage:
  criticalup-test auth [OPTIONS] [COMMAND]

Commands:
  set     Set the authentication token used to interact with the download server
  remove  Remove the authentication token used to interact with the download server
  help    Print this message or the help of the given subcommand(s)

Options:
  -v, --verbose...                  Enable debug logs, -vv for trace
      --log-format <LOG_FORMAT>     Which logger to use [default: default] [possible values: default, pretty, json]
      --log-level [<LOG_LEVEL>...]  Tracing directives
  -h, --help                        Print help
------
